src/middleware/github-webhook-middleware.ts

```diff
@@ -11,14 +11,16 @@ import { booleanFlag, BooleanFlags } from "config/feature-flags";
 import { emitWebhookFailedMetrics, emitWebhookPayloadMetrics, getCurrentTime } from "utils/webhook-utils";
 import { statsd } from "config/statsd";
 import { metricWebhooks } from "config/metric-names";
+import { WebhookContext } from "../routes/github/webhook/webhook-context";
 
 const warnOnErrorCodes = ["401", "403", "404"];
 
 // Returns an async function that reports errors errors to Sentry.
 // This works similar to Sentry.withScope but works in an async context.
 // A new Sentry hub is assigned to context.sentry and can be used later to add context to the error message.
-const withSentry = function(callback) {
-	return async (context) => {
+const withSentry = function (callback) {
+	return async (customContext?: CustomContext, webhookContext?: WebhookContext) => {
```

Comment:
Why is customContext optional? In the original code context is required. My understanding is:

- we should have a FF. If feature flag is true, use webhookContext
- if feature flag is off, default to context currently used. This should not be an optional param.

Reply:
I need to call this method with arguments `(undefined, webhookContext)` from new webhook receiver. If it make it mandatory, I won't be able to call with `undefined`. Regarding FF, I explained above.